### PR TITLE
frontend: allow workflow routes to be private

### PR DESCRIPTION
**Pull request by dschaller**
### Description
Allow workflows to define private routes. This will prevent the route from being discoverable via the drawer and search bar while still allowing them to be registered on the application.

**note**: this supersedes the trending property and will hide trending workflows.

### Testing Performed

manual

### CODE REVIEW

**Comment by mcutalo88** [comment]
This looks good, just cleaned up the test configuration for terminate instance.

File: frontend/packages/core/src/AppProvider/index.tsx [hunk 1]
```diff
@@ -28,13 +29,21 @@ interface ClutchAppProps {
 
 const ClutchApp: React.FC<ClutchAppProps> = ({ availableWorkflows, configuration }) => {
   const workflows = registeredWorkflows(availableWorkflows, configuration);
+  const publicWorkflows = _.cloneDeep(workflows).filter(workflow => {
+    const publicRoutes = workflow.routes.filter(route => {
+      return !(route?.private !== undefined ? route.private : false);
+    });
+    workflow.routes = publicRoutes; /* eslint-disable-line no-param-reassign */
+    return publicRoutes.length !== 0;
+  });
```
Comment (danielhochman) [inline frontend/packages/core/src/AppProvider/index.tsx:38]: comment this block

File: frontend/packages/core/src/AppProvider/workflow.tsx [hunk 2]
```diff
@@ -47,6 +47,8 @@ interface Route {
   path: string;
   /** Properties required by the Component that are set only via the config. */
   requiredConfigProps?: string[];
+  /** Is the workflow discoverable via search and drawer navigation. This defaults to false. */
+  private?: boolean;
```
Comment (danielhochman) [inline frontend/packages/core/src/AppProvider/workflow.tsx:51]: would `hideNav` be a better var? it's not really private.

Comment (mcutalo88) [reply frontend/packages/core/src/AppProvider/workflow.tsx:51]: Sure. `discoverable`, `hidden`, `hideNav`?

**Pull request merged**